svsim/src/main/resources/simulation-driver.cpp

```diff
@@ -926,16 +933,18 @@ void simulation_main(int argc, char const **argv) {
   delete context;
 }
 
-void run_simulation(int delay) {
+int run_simulation(int delay) {
   if(!delay) {
     testbench->eval_step();
-    return;
+    return context->gotFinish();
   }
   testbench->eval();
+  if (context->gotFinish())
+    return context->gotFinish();
   context->timeInc(delay);
+  return 0;
```

Comment:
I'm wondering if it's not worth also implementing vl_fatal, but perhaps checking `gotFinish` is equivalent for our purposes?